src/pages/signup/Signup.js

```diff
@@ -1,47 +1,560 @@
+        { id: 4, title: "POLYU", selected: false }
+      ]
+    };
+  }
+
+  handleTextInputChange = e => {
+    let temploginDTO = this.state.loginDTO;
+    temploginDTO[e.target.name] = e.target.value;
+    this.setState({
+      loginDTO: temploginDTO
+    });
+  };
+
+  handlePasswordCheckChange = e => {
+    this.setState({
+      passwordCheck: e.target.value
+    });
+  };
+
+  passwordValidation = e => {
+    const reg = new RegExp(
+      "^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,20}$"
+    );
+    const isValid = reg.test(e.target.value);
+
+    if (isValid) {
+      this.setState({
+        passwordError: "",
+        validPassword: true
+      });
+    } else if (!isValid) {
+      this.setState({
+        passwordError: "8~20자의 영문 대/소문자,숫자,특수문자를 사용하세요.",
+        validPassword: false
+      });
+    } else if (e.target.value !== "") {
+      this.setState({
+        passwordCheckError:
+          "X 비밀번호가 일치하지 않습니다. 다시 입력해주세요.",
+        samePassword: false
+      });
+    } else if (e.target.value === "" && this.state.passwordCheck === "") {
+      this.setState({
+        passwordCheckError: "",
+        samePassword: false
+      });
+    }
+  };
+
+  passwordCheckValidation = e => {
+    if (
+      this.state.loginDTO.password === e.target.value ||
+      (e.target.value === "" && this.state.loginDTO.password === "")
+    ) {
+      this.setState({
+        passwordCheckError: "",
+        samePassword: false
+      });
+      if (e.target.value !== "") {
+        this.setState({
+          samePassword: true
+        });
+      }
+    } else {
+      this.setState({
+        passwordCheckError:
+          "X 비밀번호가 일치하지 않습니다. 다시 입력해주세요.",
+        samePassword: false
+      });
+    }
+  };
+
+  handleCheckAllBox = () => {
+    this.setState({
+      checkAll: !this.state.checkAll
+    });
+    if (this.state.checkAll === false) {
+      this.setState({
+        checkUse: true,
+        checkData: true
+      });
+    } else {
+      this.setState({
+        checkUse: false,
+        checkData: false
+      });
+    }
+  };
+
+  handleCheckUseBox = () => {
+    this.setState({
+      checkUse: !this.state.checkUse
+    });
+    if (this.state.checkUse === false && this.state.checkData === true) {
+      this.setState({
+        checkAll: true
+      });
+    } else {
+      this.setState({
+        checkAll: false
+      });
+    }
+  };
+
+  handleCheckDataBox = () => {
+    this.setState({
+      checkData: !this.state.checkData
+    });
+    if (this.state.checkUse === true && this.state.checkData === false) {
+      this.setState({
+        checkAll: true
+      });
+    } else {
+      this.setState({
+        checkAll: false
+      });
+    }
+  };
+
+  resetThenSet = (id, title) => {
+    var unilist = this.state.options;
+    unilist.forEach(item => (item.selected = false));
+    unilist[id].selected = true;
+    let temploginDTO = this.state.loginDTO;
```

Comment:
```suggestion
    let tempLoginDTO = this.state.loginDTO;
```
camelCase 잘 해두는게 좋을듯! 근데 굳이 temp 를 사용하는 이유가 있는건가..?